File: backend/diet_generation/enums/meal_type.py
from enum import Enum


class MealType(Enum):
    BREAKFAST = ("breakfast", 1)
    MORNING_SNACK = ("morning_snack", 2)
    LUNCH = ("lunch", 3)
    AFTERNOON_SNACK = ("afternoon_snack", 4)
    DINNER = ("dinner", 5)
    EVENING_SNACK = ("evening_snack", 6)

    @property
    def str_value(self) -> str:
        return self.value[0]

    @property
    def meal_order(self) -> int:
        return self.value[1]

    @classmethod
    def _missing_(cls, value):
        if isinstance(value, str):
            for member in cls:
                if member.str_value == value.lower():
                    return member
        return super()._missing_(value)

    def __str__(self) -> str:
        return self.str_value

File: frontend/lib/models/diet_generation/meal_type.dart
import 'package:flutter/material.dart';

enum MealType {
  breakfast(0, 'breakfast', 'Breakfast', Icons.egg_alt),
  morningSnack(1, 'morning_snack', 'Morning Snack', Icons.free_breakfast),
  lunch(2, 'lunch', 'Lunch', Icons.rice_bowl),
  afternoonSnack(3, 'afternoon_snack', 'Afternoon Snack', Icons.cookie),
  dinner(4, 'dinner', 'Dinner', Icons.dinner_dining),
  eveningSnack(5, 'evening_snack', 'Evening Snack', Icons.bakery_dining);

  final int value;
  final String nameStr;
  final String displayName;
  final IconData icon;

 const MealType(this.value, this.nameStr, this.displayName, this.icon);

  String toJson() => nameStr;

  int toInt() => value;
  IconData toIcon() => icon;

  static MealType fromJson(String value) {
    return MealType.values.firstWhere(
      (e) => e.nameStr == value,
      orElse: () => throw ArgumentError('Unknown meal type: $value'),
    );
  }
}
